backend/main.py

```
@@ -9,46 +1,343 @@
-    format="%(asctime)s - %(name)s - %(levelname)s - %(message)s",
-)
+AppConfig = _import_app_config()
+
+
+# Try to import the auto RAG system
+def _import_auto_rag_system() -> Optional[Any]:
+    """Import AutoRAGSystem from available module paths."""
+    for module_path in ["backend.core.auto_rag", ".core.auto_rag", "core.auto_rag"]:
+        try:
+            module = importlib.import_module(module_path)
+            return getattr(module, "AutoRAGSystem")
```

Ruff (0.12.2)

Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)